src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,49 +1,111 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private int size;
+    private Object[] arrayList;
+
+    public ArrayList() {
+        arrayList = new Object[DEFAULT_CAPACITY];
+    }
+
+    public Object[] toArray(List<T> list) {
+        Object[] listToArray = new Object[list.size()];
+        for (int i = 0; i < listToArray.length; i++) {
+            listToArray[i] = list.get(i);
+        }
+        return listToArray;
+    }
 
     @Override
     public void add(T value) {
-
+        ensureCapacity();
+        arrayList[size++] = value;
     }
 
     @Override
     public void add(T value, int index) {
-
+        checkIndex(index);
+        ensureCapacity();
+        System.arraycopy(arrayList, index,
+                arrayList, index + 1,
+                size++ - index);
+        arrayList[index] = value;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        Object[] listArr = toArray(list);
```

Comment:
the name of the variable is not very good( let's not shorten words

also, do we need to make an array at all here?
can we get value from list?